### PR TITLE
Math magicians: events

### PR DESCRIPTION
**Pull request by Hisoka37**
# Hi There ✌️:

## In this PR I : 
- **Create a new directory** called `logic` inside your project
- **Copy the files** `calculate.js` and `operate.js` into your `logic/` directory
- **Install** [big.js](https://www.npmjs.com/package/big.js) using NPM

### CODE REVIEW

File: src/componenets/Calculator.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      const result = operate(previousOperation, currentOperation, operation);
+      setpreviousOperation(result);
+      setcurrentOperation('0');
+      setNewOperation(operation);
+    }
+  };
+
+  const handlePercentageClick = () => {
+    setcurrentOperation(Big(currentOperation).mod(100).toString());
+  };
```
Comment (gemmen29) [inline src/componenets/Calculator.js]: - [ ] Kindly fix the operation of `%`. It should be a regular operation like:
```
firstNumber % secondNumber = result
```
the second number shouldn't be 100. The user will enter it.

### Hint
There is no need for a particular case to handle this operation as it's already handled in the provided files `calculate.js`, and `operate.js`.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @Hisoka37, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/componenets/Calculator.js (hunk 1 above)
Comment (Hisoka37) [reply src/componenets/Calculator.js]: Hi @gemmen29 Can you help please, I don't understand what you mean I already give it a module sign it's `mod(100)`

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @Hisoka37 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done working on this milestone so far
- Excellent communication with the reviewer

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Hisoka37** [comment]
@gemmen29 @MahmoudBakr23 , Thank you so much ❤️

**Pull request merged**